benchmarks/benchmarks/ci/esmf_regridder.py

```diff
@@ -363,3 +330,82 @@ def time_lazy_regridding(self, cache):
     def time_regridding_realisation(self, cache):
         assert self.result.has_lazy_data()
         _ = self.result.data
+
+
+class TimeRegridderIO(MultiGridCompare):
+    params = [MultiGridCompare.params, ["mesh_to_grid", "grid_to_mesh"]]
+    param_names = MultiGridCompare.param_names + ["regridder type"]
+
+    def setup_cache(self):
+        from esmf_regrid.experimental.io import save_regridder
+
+        SYNTH_DATA_DIR = Path().cwd() / "tmp_data"
+        SYNTH_DATA_DIR.mkdir(exist_ok=True)
+
+        for tp in self.params[0]:
+            (
+                lon_bounds,
+                lat_bounds,
+                n_lons_src,
+                n_lats_src,
+                n_lons_tgt,
+                n_lats_tgt,
+                _,
+                alt_coord_system,
+            ) = self.get_args(tp)
+            src_grid = _grid_cube(
+                n_lons_src,
+                n_lats_src,
+                lon_bounds,
+                lat_bounds,
+                alt_coord_system=alt_coord_system,
+            )
+            tgt_grid = _grid_cube(
+                n_lons_tgt,
+                n_lats_tgt,
+                lon_bounds,
+                lat_bounds,
+                alt_coord_system=alt_coord_system,
+            )
+            src_mesh_cube = _gridlike_mesh_cube(
+                n_lons_src,
+                n_lats_src,
+            )
+            tgt_mesh_cube = _gridlike_mesh_cube(
+                n_lons_tgt,
+                n_lats_tgt,
+            )
+
+            mesh_to_grid_regridder = MeshToGridESMFRegridder(src_mesh_cube, tgt_grid)
+            grid_to_mesh_regridder = GridToMeshESMFRegridder(src_grid, tgt_mesh_cube)
+
+            source_file_m2g = str(
+                SYNTH_DATA_DIR.joinpath(f"source_{tp}_mesh_to_grid.nc")
+            )
+            source_file_g2m = str(
+                SYNTH_DATA_DIR.joinpath(f"source_{tp}_grid_to_mesh.nc")
+            )
+
+            save_regridder(mesh_to_grid_regridder, source_file_m2g)
+            save_regridder(grid_to_mesh_regridder, source_file_g2m)
```

Comment:
Stick this in a loop of `params[1]` too. You're already half way there with your regeneration of `self.source_fle` in `setup()`.